src/core/init/stall_avoider.ts

```diff
@@ -168,15 +182,46 @@ export default function StallAvoider(
 
   let prevFreezingState : { attemptTimestamp : number } | null;
 
-  return clock$.pipe(
+  /**
+   * If we're rebuffering waiting on data of a "locked stream", seek into the
+   * Period handled by that stream to unlock the situation.
+   */
+  const unlock$ = lockedStream$.pipe(
+    withLatestFrom(clock$),
+    tap(([lockedStreamEvt, tick]) => {
+      // TODO(PaulB) also skip when the user's wanted speed is set to `0`, as we
```

Comment:
Is it something  that u plan to do after the merge of this PR ?

Reply:
The problem is that this is now not possible.

We can now at which `playbackRate` the video element currently plays, but it will be influenced by the fact that the RxPlayer set it to `0` when rebuffering.

Here, we want to catch when the user is setting it to `0`.

We could use the `speed$` BehaviorSubject for that (whose purpose is exactly to communicate the last wanted speed), but for now it is defined as an Observable in `Init`, so no `getValue` method is available until we make it a `BehaviorSubject` explicitely there - and I'm not that comfortable with that for now (considering it is supposed to be read-only in the Init, making it a BehaviorSubject just to authorize synchronous access also gives the right to emit through it, sadly. I don't want to do the latter because it implicitely gives the idea to the code's reader that the Init can update the wanted speed, which it doesn't).

We could also develop in the clock something like the `internal-seek` for `internal-speed`, but this is slightly different because `internal-speed` would be about the temporarily disabling of speed updates whereas an `internal-seek` only happends at a point in time. There we would also need to store the wanted speed (so something like `speed$`) in the clock.

For now this subject is too complex, and performing a seek when rebuffering close to a locked Period when the playback is set to `0` is not all that annoying, so I let it pass for a future release. The time for us refactoring some other bricks like the clock to see where we could go with that.

Reply:
Oh ok understood! thanks:)